src/uwtools/api/config.py

```diff
@@ -68,10 +97,35 @@ def translate(
     return False
 
 
-def validate(input_file: DefinitePath, input_format: str, schema_file: DefinitePath) -> bool:
+def validate(
+    schema_file: DefinitePath, config: Optional[Union[dict, _YAMLConfig, OptionalPath]] = None
+) -> bool:
     """
     ???
     """
-    if input_format == _FORMAT.yaml:
-        return _validate(config_file=input_file, schema_file=schema_file)
-    return False
+    if isinstance(config, dict):
+        cfgobj = _YAMLConfig(empty=True)
+        cfgobj.update(config)
+    elif isinstance(config, _YAMLConfig):
+        cfgobj = config
+    else:
+        cfgobj = _YAMLConfig(config_file=config)
+    return _validate_yaml_config(schema_file=schema_file, config=cfgobj)
```

Comment:
`validate()`, as extended for API use, can operate on an in-memory `dict` or `Config` objects, as well as the existing support for file/stream IO. The `YAMLConfig` class is used for validating a `dict` object.